docker_release/main.py

```diff
@@ -46,6 +52,18 @@ def _get_docker_image_name(docker_image_dir):
     return (path.basename(path.dirname(docker_image_dir)), path.basename(docker_image_dir))
 
 
+def _check_labelled(docker, image, repo):
+    sha = _get_current_sha(repo)
+    inspect = docker.inspect_image(image)
+    labels = inspect['Config']['Labels']
+    try:
+        label_hash = labels[GIT_HASH]
+        if label_hash != sha:
+            raise UserMessageException('Image %s hash mismatch in label %s. Expected %s, found %s' % (image, GIT_HASH, sha, label_hash))
+    except KeyError as e:
+        raise UserMessageException('Image %s missing %s label' % (image, e))
```

Comment:
If we're going for universality, we should use  vcs-ref from the [labeling schema](http://label-schema.org/rc1/#build-time-labels)